src/reactive/duplicity.py

```diff
@@ -56,6 +56,7 @@ def install_duplicity():
     fetch.apt_install("python-paramiko")
     fetch.apt_install("python-boto")
     fetch.apt_install("lftp")
+    os.system("pip install azure-storage-blob")
```

Comment:
**nitpick (blocking):** 

Is this always required? Also `os.system` is not a recommended way to run pip install. I suggest move dependency to requirements.txt so the package will be included when charm is building.

Reply:
actually i tried adding it in requirements.txt at the first place, but it's not working as expected. i am doing local charmcraft pack and then deploy. 
It give the "package azure not found" error.
Any advice ?

Reply:
After checking the behavior of charm.

The charm use subprocess to call `duplicity`, which is installed by apt. So the `azure-storage-blob` has to be install in system level python.

Sorry for the misleading.

But here we should not use `os.system`, we can use `subprocess` to replace it and also I will make it as a separate function like `def install_in_system_python` for better testable in the unittest.

Reply:
added as adviced.

Reply:
Can you use 
```
from charmhelpers.fetch.python import packages
...
packages.pip_install("...")
```
instead

Reply:
It is not working as expected, so for now i have to use subprocess.